### DISCUSSION TITLE
6490 Cable in "ATA mode", how to configure voip VLAN

### DISCUSSION THREAD

**Discussion by afflux** [Freetz-NG/freetz-ng#765] [Allgemein] [answered]
Hi! I have a 6490 cable (atom) lying about which I want to use in ATA mode (not sure if that's the right word) as a router on my GPON modem, so DHCP client on LAN 1 (untagged). My ISP is also offering VoIP on a tagged VLAN (also with DHCP).

I have seen a config interface on AVMs web interface in screenshots, but it seems like this was on non-cable boxes: 
![NetAachen-FritzBox-Telefonie-Anschlusseinstellungen](https://user-images.githubusercontent.com/768184/224839911-3207ad83-02ec-4dcb-b6f7-bc066d3b82f4.png)

When I switch my 6490 cable box to go online via LAN 1, I don't see these options in voip settings.

I'm thinking I might need to manually configure the VLAN in ar7.cfg but it's not clear to me which section to look into (maybe VCCS?).
Any suggestions welcome.

**Comment by afflux** [comment, marked as answer]
Made a bunch of changes that seem to do the trick.

In /var/flash/ar7.cfg:
```
ar7cfg.sync_groups["sync_cable"].enabled = no
ar7cfg.sync_groups["sync_ata"].enabled = yes
ar7cfg.vccs[2 /* the one vccs section without ifname eth_udma */].connections = "internet", "voip"
ar7cfg.dslifaces["voip"].vlancfg {
	vlanencap = vlanencap_fixed_prio;
	tagtype = vlantagtype_customer;
	vlanid = 6;
	vlanprio = 0;
	tos = 0;
}
ar7cfg.dslglobalconfig.pppoeiface = "eth0";

/* not sure if these are really needed: */
ar7cfg.dslifaces["voip"].etherencapcfg.class_identifier = "6490retail"; /* this is transmitted as DHCP option 60 */
ar7cfg.dslifaces["voip"].stay_always_online = no;
ar7cfg.targets["voip"].remoteauth = pppcfg_authtype_none;
```

In /var/flash/voip.cfg, set `sipiface = sipiface_voip` for all registered numbers.

Looks ok on the wire, will be able to test this setup in the next few weeks.

**Reply by afflux** [reply]
appears to work fine

**Reply by darksapien** [reply]
Glad you got this working, I am in a similar situation where I need to set the VLAN ID to 200, Does your box now show the screenshot above?

**Reply by afflux** [reply]
No, that UI is disabled client-side on cable boxes.

**Comment by fda77** [comment]
If the webif pages are still present in the firmware,  likely /usr/www/a*/ it should be possible with a patch to enable them again

I picked "vlanencap_fixed_prio" for a quick check:

```
 grep vlanencap_fixed_prio  usr/www/ -r
usr/www/avm/fon_num/phoneline.lua:      cmtable.add_var(saveset,"connection_voip:settings/vlanencap","vlanencap_fixed_prio")
```

**Reply by afflux** [reply]
They are indeed still there, that's how I figured out what to do in the first place. I didn't test it, but this patch should enable the UI:

```patch
--- usr/www/all/fon_num/phoneline.lua.bak       2023-04-11 16:14:29.301107400 +0200
+++ usr/www/all/fon_num/phoneline.lua   2023-04-11 16:15:25.281107400 +0200
@@ -427,7 +427,7 @@
 local show = {}
 show.t38 = config.T38
 show.sipping = true
-show.vlan = not (config.is_cable or (config.LTE and not config.multiwan_lte))
+show.vlan = true
 show.use_dqos = config.is_cable_retail
 if box.query("box:settings/voip_2ndPVC_gui_hidden") == "1" then
 show.scndpvc = false
@@ -460,7 +460,7 @@
 result.scndpdn = data_scndpdn{}
 end
 result.multiwan_lte = config.multiwan_lte
-result.is_cable = config.is_cable
+result.is_cable = false
 return result
 end
 end

```

**Comment by fda77** [comment]
@afflux: I hope it works, please test: 4913a8e35
Im not sure if other devices or fritzos versions should be unlocked, i did not want to test a long time :)

**Reply by afflux** [reply]
UI shows, displays my current config. Changing VLAN id reflects correctly in ar7.cfg. Looks good to me!
